sdk/cosmos/azure_data_cosmos/src/retry_policies/resource_throttle_retry_policy.rs

```diff
@@ -0,0 +1,440 @@
+
+    /// Helper function to create a mock RawResponse with a given status code
+    fn create_mock_response(status: StatusCode) -> azure_core::Result<RawResponse> {
+        use typespec_client_core::http::headers::Headers;
+
+        // Create headers
+        let mut headers = Headers::new();
+        headers.insert("content-type", "application/json");
+
+        azure_core::Result::from(Ok(RawResponse::from_bytes(
+            status,
+            headers,
+            r#"{"id":12,"name":"Too Many Requests"}"#,
+        )))
+    }
+    #[tokio::test]
+    async fn test_retry_policy_handles_429_status() {
+        // Create a retry policy with 3 max retries, 100 second max wait time, and backoff factor of 2
+        let policy = ResourceThrottleRetryPolicy::new(3, 100, 2);
+
+        // Simulate a 429 TooManyRequests response
+        let response_429 = create_mock_response(StatusCode::TooManyRequests);
+
+        // First retry attempt
+        let result1 = policy.should_retry(&response_429).await;
+        match result1 {
+            RetryResult::Retry { after } => {
+                assert!(after > Duration::ZERO, "Should have backoff time");
+                println!("First retry - backoff time: {:?}", after);
+            }
+            RetryResult::DoNotRetry => panic!("Should retry on first 429 response"),
+        }
+
+        // Second retry attempt
+        let result2 = policy.should_retry(&response_429).await;
+        let backoff2 = match result2 {
+            RetryResult::Retry { after } => {
+                assert!(after > Duration::ZERO, "Should have backoff time");
+                after
+            }
+            RetryResult::DoNotRetry => panic!("Should retry on second 429 response"),
+        };
+
+        // Extract backoff1 for comparison
+        let backoff1 = match result1 {
+            RetryResult::Retry { after } => after,
+            _ => panic!("Expected retry result"),
+        };
+
+        assert!(
+            backoff2 >= backoff1,
+            "Backoff should increase with exponential backoff"
+        );
+        println!("Second retry - backoff time: {:?}", backoff2);
+
+        // Third retry attempt
+        let result3 = policy.should_retry(&response_429).await;
+        match result3 {
+            RetryResult::Retry { after } => {
+                assert!(after > Duration::ZERO, "Should have backoff time");
+                println!("Third retry - backoff time: {:?}", after);
+            }
+            RetryResult::DoNotRetry => panic!("Should retry on third 429 response"),
+        }
+
+        // Fourth attempt should NOT retry (exceeded max_attempt_count of 3)
+        let result4 = policy.should_retry(&response_429).await;
+        match result4 {
+            RetryResult::DoNotRetry => {
+                println!("Fourth attempt - should not retry");
+            }
+            RetryResult::Retry { .. } => panic!("Should NOT retry after exceeding max attempts"),
+        }
```

Comment:
This can be an `assert_eq!` I think:


```suggestion
        assert_eq!(result4, RetryResult::DoNotRetry);
```

Reply:
Updated.